### ISSUE TITLE
howto read image data from stdin

### ISSUE THREAD

**Issue by ewelot** [libvips/libvips#1915]
I kindly ask for help on the topic of reading image data from stdin (in a python program). I have tried something like
```
    source = pyvips.Source.new_from_descriptor(sys.stdin.fileno())
    img1 = pyvips.Image.new_from_source(source, '')
```
but this does not succeed. Probably I need to provide more information about the image data (which loader to use?) to the program. Or use another approach?

I am aware of this [article](https://libvips.github.io/libvips/2019/11/29/True-streaming-for-libvips.html) but failed to adopt it. I must confess that I am a both a newbie on libvips and not very familiar with python coding either ...

Would be great if you could provide a simple example.
Thanks in advance.

**Comment by jcupitt** [comment]
This works for me with 8.11:

```
#!/usr/bin/python3
  
import sys
import pyvips

source = pyvips.Source.new_from_descriptor(sys.stdin.fileno())
image = pyvips.Image.new_from_source(source, "")
target = pyvips.Target.new_to_descriptor(sys.stdout.fileno())
image.write_to_target(target, ".jpg")
```

Then:

```
$ ./stdio.py < ~/pics/k2.jpg >x.jpg
```

I;ll try 8.10 as well.

**Comment by ewelot** [comment]
That is exactly what I did. Well, just a tiny difference: I tried loading a PPM and in this case I get
```
Traceback (most recent call last):
  File "./stdio.py", line 7, in <module>
    image = pyvips.Image.new_from_source(source, "")
  File "/home/lehmann/.local/lib/python3.7/site-packages/pyvips/vimage.py", line 404, in new_from_source
    raise Error('unable to load from source')
pyvips.error.Error: unable to load from source
  VipsForeignLoad: source is not in a known format
```
The program runs fine when using JPEG image from stdin. Is it possible to add support for streaming of PPM (and FITS) as well?

**Comment by jcupitt** [comment]
The code was all there, it was just missing the link to the sniffer. Fixed in 8.10.3, thanks for finding this.

FITS is tricky: cfitsio only supports file IO, as far as I know.

**Comment by ewelot** [comment]
I can confirm it is working now with PPM images as well. Thanks!
